roles/openshift_repos/tasks/main.yaml

```diff
@@ -28,21 +28,26 @@
   # "centos-release-openshift-origin" package which configures the repository.
   # This task matches the file names provided by the package so that they are
   # not installed twice in different files and remains idempotent.
-  - name: Configure origin gpg keys if needed
-    copy:
-      src: "{{ item.src }}"
-      dest: "{{ item.dest }}"
-    with_items:
-    - src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
-      dest: /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-PaaS
-    - src: origin/repos/openshift-ansible-centos-paas-sig.repo
-      dest: /etc/yum.repos.d/CentOS-OpenShift-Origin.repo
-    notify: refresh cache
-    when:
+  - when:
     - ansible_os_family == "RedHat"
     - ansible_distribution != "Fedora"
     - openshift_deployment_type == 'origin'
     - openshift_enable_origin_repo | default(true) | bool
+    block:
+    - name: Configure origin repositories and gpg keys if needed
+      copy:
+        src: "{{ item.src }}"
+        dest: "{{ item.dest }}"
+      with_items:
+      - src: origin/gpg_keys/openshift-ansible-CentOS-SIG-PaaS
+        dest: /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-SIG-PaaS
+      - src: origin/repos/openshift-ansible-centos-paas-sig.repo
+        dest: /etc/yum.repos.d/CentOS-OpenShift-Origin.repo
+      notify: refresh cache
+
+    - name: Enable centos-openshift-origin-testing repository
+      command: yum-config-manager --enable centos-openshift-origin-testing
+      when: openshift_repos_enable_testing
```

Comment:
We should add a `| bool` filter to the end of this check on the odd chance we get a string.